### PR TITLE
🚀 New Upgrades

### PR DESCRIPTION
**Pull request by Flowko**
This pull request aims to enhance the overall website experience by improving existing features and introducing new ones. Currently, the integration of Nuxt 3 is not optimized to its fullest potential.

### Key Features to be Included:
- 🦄 Intuitive SVG icon usage
- 📥 Automatic API importing for Composition API, VueUse, and Components
- 🦾 Incorporation of TypeScript and comprehensive type support (primarily, with ongoing improvements planned)
- 🤖 Seamless integration with existing modules for effortless implementation
- 🧽 Enhanced code cleanliness
- 📦 Utilization of Vue 3 Composition API
- 📔 Updated Eslint config
- 👾 Extras

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -3,11 +3,10 @@ root = true
 [*]
 indent_style = space
 indent_size = 2
-end_of_line = lf
+end_of_line = crlf
```
Comment (bluebear94) [inline .editorconfig:6]: Just no.

Comment (Flowko) [reply .editorconfig:6]: sure

**Comment by Prospector** [comment]
This is something you should have consulted the Modrinth team on before doing, it is unlikely it will be possible to merge this as it'll likely have serious conflicts with active development branches, and we plan to move to our components library Omorphia shortly.

**Comment by Flowko** [comment]
yeah, i just figured to give it a try, but yeah i'll stop working on it, thanks

**Comment by Prospector** [comment]
I appreciate the work though, I hope we can make a bunch of these changes eventually.

**Pull request closed without merging**